ci.sh

```diff
@@ -37,25 +39,53 @@ python -c "import sys, struct, ssl; print('python:', sys.version); print('versio
 echo "::endgroup::"
 
 echo "::group::Install dependencies"
-python -m pip install -U pip build
+python -m pip install -U pip uv
 python -m pip --version
+uv --version
+
+# expands to 0 != 1 if MAKE_VENV is not set, if set the `-0` has no effect
+# https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02
+if [ "${MAKE_VENV-0}" == 1 ]; then
+    uv venv .venv --seed
+
+    if [ "$ON_WINDOWS" = "True" ]; then
+        source ./.venv/Scripts/activate
+    else
+        source .venv/bin/activate
+    fi
+
+    # Make sure pip is installed (pypy weirdness)
+    python -m ensurepip
+
+    # Install uv in venv
+    python -m pip install --upgrade uv
+fi
+
+if [ "$ON_WINDOWS" = "True" ]; then
+    PYTHON_PATH=$(python -c "import os, sys; print(os.path.dirname(sys.executable))")
+else
+    PYTHON_PATH=$(which python)
+fi
+
+uv pip install --python="$PYTHON_PATH" build
```

Comment:
Is this needed?

> When running pip sync or pip install, uv will search for a virtual environment in the following order:
> 1. An activated virtual environment based on the VIRTUAL_ENV environment variable.
> 2. An activated Conda environment based on the CONDA_PREFIX environment variable.
> 3. A virtual environment at .venv in the current directory, or in the nearest parent directory.

https://github.com/astral-sh/uv?tab=readme-ov-file#advanced-usage

Reply:
Removed in 57b8d2f